lib/modules/manager/maven/lockfile.spec.ts

```diff
@@ -0,0 +1,152 @@
+      try {
+        tempFilePath = createDummyLockfile('simpleprojectWithLockfile');
+        mockLockfileChangedInGit();
+        GlobalConfig.set({
+          localDir: join(
+            'lib/modules/manager/maven/__fixtures__/simpleprojectWithLockfile'
+          ),
+        });
+        const result = await updateArtifacts({
+          packageFileName:
+            'lib/modules/manager/maven/__fixtures__/simpleprojectWithLockfile/pom.xml',
+          updatedDeps: [{ datasource: 'maven' }],
+          newPackageFileContent: '{}',
+          config: {},
+        });
+        expect(result).not.toBeNull();
+        expect(result).toHaveLength(1);
+        expect(result![0].file?.path).toBe('lockfile.json');
+      } finally {
+        if (tempFilePath) {
+          await deleteDummyLockfile(tempFilePath);
+        }
+      }
+    });
+
+    it('only lockfiles which are modified are added', async () => {
+      let tempFilePath: string | undefined = undefined;
+      try {
+        tempFilePath = createDummyLockfile('simpleprojectWithLockfile');
+        mockLockfileCreatedInGit();
+        GlobalConfig.set({
+          localDir: join(
+            'lib/modules/manager/maven/__fixtures__/simpleprojectWithLockfile'
+          ),
+        });
+        const result = await updateArtifacts({
+          packageFileName:
+            'lib/modules/manager/maven/__fixtures__/simpleprojectWithLockfile/pom.xml',
+          updatedDeps: [{ datasource: 'maven' }],
+          newPackageFileContent: '{}',
+          config: {},
+        });
+        expect(result).not.toBeNull();
+        expect(result).toHaveLength(0);
+      } finally {
+        if (tempFilePath) {
+          await deleteDummyLockfile(tempFilePath);
+        }
+      }
+    });
+
+    it('parent/child project work', async () => {
+      let tempFilePath: string | undefined = undefined;
+      try {
+        tempFilePath = createDummyLockfile(
+          'parent_child/simpleprojectWithLockfile'
+        );
+        mockLockfileChangedInGit('simpleprojectWithLockfile');
+        GlobalConfig.set({
+          localDir: join(
+            'lib/modules/manager/maven/__fixtures__/parent_child/'
+          ),
+        });
+        const result = await updateArtifacts({
+          packageFileName:
+            'lib/modules/manager/maven/__fixtures__/parent_child/simpleprojectWithLockfile/pom.xml',
+          updatedDeps: [{ datasource: 'maven' }],
+          newPackageFileContent: '{}',
+          config: {},
+        });
+        expect(result).not.toBeNull();
+        expect(result).toHaveLength(1);
+      } finally {
+        if (tempFilePath) {
+          await deleteDummyLockfile(tempFilePath);
+        }
+      }
+    });
+  });
+});
+
+function createDummyLockfile(subproject: string) {
+  const tempFilePath = join(
+    __dirname,
+    '__fixtures__',
+    subproject,
+    'lockfile.json'
+  );
+  fs.writeFileSync(tempFilePath, '{}', 'utf8');
```

Comment:
don't create files inside fixture folder, copy files to a temp folder instead or mock util/fs like we do on other magager